### PR TITLE
Use MallocMapping for DispatchSemanticsAction data

### PR DESCRIPTION
**Pull request by cbracken**
In the Windows embedder, the data parameter of
FlutterWindowsEngine::DispatchSemanticsAction was a vector of bytes.
Here, we switch it to use MallocMapping since that is the type of the
data paramater on the DispatchAccessibilityAction method of
AccessibilityBridgeDelegate, which calls this method. This avoids an
extra type conversion.

This is a cleanup patch with no functional changes, but makes this
call very slightly more performant in the follow-up events patch.

Issue: https://github.com/flutter/flutter/issues/77838

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by justinmc — APPROVED** [review]
LGTM

I'm no C++ expert though, so I'd get a second review if you have any doubts about switching to MallocMapping here.

**Comment by cbracken** [comment]
> if you have any doubts about switching to MallocMapping here.

It was me who erroneously made the type `std::vector<uint_8>` in the first place. I was thinking that a simple byte vector would aid readability, but hadn't noticed that the caller of this method (which is always AccessibilityBridgeDelegate, at least in the macOS/Win32 embedder), already has an `fml::MallocMapping`.

**Comment by gw280** [comment]
Bit late to the party but LGTM

**Pull request merged**
